pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -144,6 +144,34 @@ func NewCloudCIDRAllocator(client clientset.Interface, cloud cloudprovider.Inter
 		DeleteFunc: nodeutil.CreateDeleteNodeHandler(ca.ReleaseCIDR),
 	})
 
+	nwInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(originalObj interface{}) {
+			nwOr, isNetwork := originalObj.(*networkv1.Network)
+			if !isNetwork {
+				klog.Errorf("Received unexpected object: %v", originalObj)
+				return
+			}
+			nw := nwOr.DeepCopy()
```

Comment:
why do you DeepCopy? these are very expensive operations and will only be needed if you are going to mutate the object.
We should avoid mutating objects as much as we can, it may be needed if you want to `Update`, but if you can use `Patch`instead that will be much better in different dimensions

Reply:
This is a copy-paste from`nodeutil.CreateAddNodeHandler`. Will remove here.